dart/lib/src/locator.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:ffi';
+import 'dart:io';
+
+/// Attempts to locate the MinifyHtml dynamic library.
+///
+/// Throws [MinifyHtmlLocatorError] if the dynamic library could not be found.
+DynamicLibrary loadDynamicLibrary() {
+  String locate(String libName) {
+    final dir = '$_minifyhtmlToolDir$libName';
+    final value = _packageRootUri(Platform.script.resolve('./')) ??
+        _packageRootUri(Directory.current.uri);
+    if (value != null) {
+      return value.resolve(dir).toFilePath();
+    } else {
+      throw MinifyHtmlLocatorError(
+        'MinifyHtml library not found. Did you run `$invocationString`?',
+      );
+    }
+  }
+
+  if (Platform.isIOS) {
+    return DynamicLibrary.process();
+  } else if (Platform.isMacOS) {
+    return DynamicLibrary.open(locate(appleLib));
+  } else if (Platform.isWindows) {
+    return DynamicLibrary.open(locate(windowsLib));
+  } else if (Platform.isLinux) {
+    return DynamicLibrary.open(locate(linuxLib));
+  } else if (Platform.isAndroid) {
+    return DynamicLibrary.open(linuxLib);
```

Comment:
Why not call `locate` for Android?

Reply:
In android the library is packaged into the apk in jniLibs/<arch> directory. The library can then simply be loaded by name.